src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/PopOperation.java

```diff
@@ -60,14 +60,18 @@ public boolean updateDocument(ObjectNode doc, UpdateTargetLocator targetLocator)
     boolean changes = false;
     for (PopAction action : actions) {
       final String path = action.path;
-      JsonNode node = doc.get(path);
-      ArrayNode array;
+      UpdateTarget target = targetLocator.findIfExists(doc, path);
 
-      if (node == null) { // No such property? Fine, no change
-        ;
-      } else if (node.isArray()) { // Already array? To modify unless empty
-        if (!node.isEmpty()) {
-          array = (ArrayNode) node;
+      JsonNode value = target.valueNode();
+      // If target does not match, nothing to do; not an error
+      if (value == null) {
+        continue;
+      }
+
+      // Otherwise must be an array
+      if (value.isArray()) { // Already array? To modify unless empty
+        if (!value.isEmpty()) {
```

Comment:
Interesting we have handled empty array. I just tested it with insert and find, it works!

Reply:
Ok good -- I thought it should but is good to confirm. Mongo array operators are pretty useful.